docs/whitepaper/whitepaper.md

```diff
@@ -0,0 +1,591 @@
+
+### 4.1.3 Component Diagram
+
+Source file: `./docs/architecture/component-diagram.puml`
+
+![Component DiagramHyperledger Blockchain Integration Framework](https://www.plantuml.com/plantuml/png/0/ZLN1Rjim3BthAmXVrWQhiVGO546pPaK7x32i1NOPCCWownYH9LTKbdX9_tsKx2JsihRBIT9xV7mYAUUQl7H-LMaXVEarmesjQclGU9YNid2o-c7kcXgTnhn01n-rLKkraAM1pyOZ4tnf3Tmo4TVMBONWqD8taDnOGsXeHJDTM5VwHPM0951I5x0L02Cm73C1ZniVjzv9Gr85lTlIICqg4yYirIYDU1P2PiGKvI6PVtc8MhdsFQcue5LTM-SnFqrF4vWv9vkhKsZQnbPS2WPZbWFxld_Q4jTIQpmoliTj2sMXFWSaLciQpE-hmjP_ph7MjgduQ7-BlBl6Yg9nDNGtWLF7VSqsVzHQTq8opnqITTNjSGUtYI6aNeefkS7kKIg4v1CfPzTVdVrLvkXY7DOSDsJTU-jaWGCQdT8OzrPPVITDJWkvn6_uj49gxVZDWXm-HKzIAQozp3GEyn_gEpoUlfs7wb39NYAAYGWrAXwQeTu4XliWhxGaWkJXEAkTM7lB3evzZq2S1yO2ACAysekBsF49N5t9ed1OI8_JQOS-CxpRnaSYte6n7eE86VC6O0OyFOoP_PJ36Ao3oZfc7QOyRRdcU1H3CZo-3SWQaAQ9HBEgCdxNzX7EVgEpu2rKZ9s7N54BJHwDyFACBRwFviuXJOCj4OVtUSUN-jlpvT5pR-B3YFFiRBXskc7_1vClsmwFudyTzpAzPVwoCpzYxwH2ErJuz54PcieDEO3hLx3OtbTmgaz1qSv4CavWjqjJk-LbceuI7YB-26_ONBf_1SCjXMto8KqvahN3YgEm5litq-cC_W7oK8uX_aBM0K5SSvNu7-0F "Component DiagramHyperledger Blockchain Integration Framework")
+
+### 4.1.4 Class Diagram
+
+### 4.1.5 Sequence Diagram - Transactions
+
+TBD
+
+## 4.2 Transaction Protocol Specification
+
+### 4.2.1 Handshake Mechanism
+
+TBD
+
+### 4.1.9 Transaction Protocol Negotiation
+
+Participants in the transaction must have a handshake mechanism where they agree on one of the supported protocols to use to execute the transaction. The algorithm looks an intersection in the list of supported algorithms by the participants.
+
+Participants can insist on a specific protocol by pretending that they only support said protocol only.
+Protocols can be versioned as the specifications mature.
+Adding new protocols must be possible as part of the plugin architecture allowing the community to propose, develop, test and release their own implementations at will.
+The two initially supported protocols shall be the ones that can satisfy the requirements for Fujitsu’s and Accenture’s implementations respectively.
+Means for establishing bi-directional communication channels through proxies/firewalls/NAT wherever possible
+
+## 4.3 Plugin Architecture
+
+Since our goal is integration, it is critical that `BIF` has the flexibility of supporting most ledgers, even those that don't exist today.
```

Comment:
+1, this is important and something we should emphasize to the broader community.

Reply:
Thank you!